api/app/controllers/review.controller.ts

```diff
@@ -444,6 +445,75 @@ export const getAllReviewsForAdmin = async (req: Request, res: Response) => {
   }
 };
 
+export const getReviewsByPaymentStatus = async (
+  req: Request,
+  res: Response
+) => {
+  const status = (req.query.status as string)?.toLowerCase();
+
+  if (!status) {
+    return res.status(400).json({
+      message: "status of 'paid' or 'unpaid' is required",
+    });
+  }
+
+  const validStatuses = ["paid", "unpaid"];
+
+  if (!validStatuses.includes(status)) {
+    return res.status(400).json({
+      message: "status must be either 'paid' or 'unpaid'",
+    });
+  }
+
+  try {
+    const mergedReviews = await Review.findAll({
+      where: { mergedAt: { [Op.not]: null } },
+    });
+    const allCreditTransactions = await Transaction.findAll({
+      where: { transactionType: TRANSACTION_TYPE.CREDIT },
+    });
+
+    const creditTransactionReviewIds = allCreditTransactions.map(
+      (transaction) => transaction.reviewId
+    );
+    const unpaidMergedReviews = mergedReviews.filter(
+      (review) => !creditTransactionReviewIds.includes(review.id)
+    );
+    const paidMergedReviews = mergedReviews.filter((review) =>
+      creditTransactionReviewIds.includes(review.id)
+    );
+
+    let response: {
+      totalItems: number;
+      data: Review[];
+    } = {
+      totalItems: 0,
+      data: [],
+    };
+
+    if (status === "paid") {
+      response = {
+        totalItems: paidMergedReviews.length,
+        data: paidMergedReviews,
+      };
+      return res.status(200).json(response);
+    } else if (status === "unpaid") {
+      response = {
+        totalItems: unpaidMergedReviews.length,
+        data: unpaidMergedReviews,
+      };
+      return res.status(200).json(response);
+    }
```

Comment:
Also instead of creating two separate arrays for paid and unpaid reviews, we can create a single array based on the status query parameter. ( ways we can optimise)

Reply:
Can you elaborate on how to do this?